source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using GreenEnergyHub.Charges.Domain.Dtos.ChargeCommands.Validation;
+
+namespace GreenEnergyHub.Charges.Infrastructure.Core.Cim.ValidationErrors
+{
+    public static class CimValidationErrorTextTemplateMessages
+    {
+        /// <summary>
+        /// Errormessage for <see cref="ValidationRuleIdentifier.StartDateValidation"/>
+        /// using <see cref="CimValidationErrorTextToken.ChargeStartDateTime"/>
+        /// </summary>
```

Comment:
Just a thought: How about making these relationships stronger by e.g. applying them as attributes here and then e.g. finding the template text by attribute instead of maintaining a huge switch somewhere?

Perhaps something like:
```suggestion
        [ErrormessageFor(ValidationRuleIdentifier.StartDateValidation)]
        [Tokens(CimValidationErrorTextToken.ChargeStartDateTime)]
```